Description: Add armv6 asm for memory barriers.
 armv6 doesn't have dmb but it does have a coprocessor instruction that
 provides the same function.
Author: Peter Michael Green <plugwash@raspbian.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- trafficserver-5.0.1.orig/plugins/header_rewrite/lulu.h
+++ trafficserver-5.0.1/plugins/header_rewrite/lulu.h
@@ -45,6 +45,10 @@ char* getIP(sockaddr const* s_sockaddr,
 #define mb()  __asm__ __volatile__ ( "sync" : : : "memory")
 #define rmb() __asm__ __volatile__ ( "sync" : : : "memory")
 #define wmb() __asm__ __volatile__ ( "" : : : "memory")
+#elif defined(__ARM_ARCH_6__)
+#define mb()  __asm__ __volatile__ ( "mcr p15, #0, r0, c7, c10, #5" : : : "memory")
+#define rmb() __asm__ __volatile__ ( "mcr p15, #0, r0, c7, c10, #5" : : : "memory")
+#define wmb() __asm__ __volatile__ ( "" : : : "memory")
 #elif defined(__arm__)
 #define mb()  __asm__ __volatile__ ( "dmb" : : : "memory")
 #define rmb() __asm__ __volatile__ ( "dmb" : : : "memory")
